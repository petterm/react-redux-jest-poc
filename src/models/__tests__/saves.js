/* eslint func-names: 'off', prefer-arrow-callback: 'off' */
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import savesReducer, {
  initialState,
  syncronize,
  syncStart,
  syncSuccess,
  syncFail,
} from '../saves';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('handles syncronize actions', function () {
  const mockSyncResponse = {
    'save 1': {
      timestamp: 1234567,
    },
    another_save: {
      timestamp: 2234567,
    },
  };
  it('handles successful syncronize action', function () {
    const store = mockStore({
      ...initialState,
    });
    const response = JSON.stringify(mockSyncResponse);
    fetch.mockResponseOnce(response);
    return store.dispatch(syncronize())
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
  it('handles failed syncronize action', function () {
    const store = mockStore({
      ...initialState,
    });
    const response = JSON.stringify({ error: 'some error message' });
    fetch.mockResponseOnce(response, { status: 500, statusText: 'Server error' });
    return store.dispatch(syncronize())
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
  it('reduces syncStart action', function () {
    expect(savesReducer(initialState, syncStart())).toMatchSnapshot();
    expect(savesReducer({
      ...initialState,
      syncError: Error('Old error'),
    }, syncStart())).toMatchSnapshot();
  });
  it('reduces syncSuccess action', function () {
    expect(savesReducer({
      ...initialState,
      syncronizing: true,
    }, syncSuccess(mockSyncResponse))).toMatchSnapshot();
  });
  it('reduces syncFail action', function () {
    expect(savesReducer(initialState, syncFail(Error('Some error')))).toMatchSnapshot();
  });
});
