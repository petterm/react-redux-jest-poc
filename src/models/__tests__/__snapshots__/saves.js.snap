exports[`handles syncronize actions handles failed syncronize action 1`] = `
Array [
  Object {
    "type": "SAVES_SYNC_START",
  },
  Object {
    "payload": [Error: Server error],
    "type": "SAVES_SYNC_FAIL",
  },
]
`;

exports[`handles syncronize actions handles successful syncronize action 1`] = `
Array [
  Object {
    "type": "SAVES_SYNC_START",
  },
  Object {
    "payload": Object {
      "another_save": Object {
        "timestamp": 2234567,
      },
      "save 1": Object {
        "timestamp": 1234567,
      },
    },
    "type": "SAVES_SYNC_SUCCESS",
  },
]
`;

exports[`handles syncronize actions reduces syncFail action 1`] = `
Object {
  "deleteError": undefined,
  "deleting": false,
  "newSaveConfig": Object {},
  "saves": Object {},
  "syncError": [Error: Some error],
  "syncronizing": false,
  "uploadError": undefined,
  "uploading": false,
}
`;

exports[`handles syncronize actions reduces syncStart action 1`] = `
Object {
  "deleteError": undefined,
  "deleting": false,
  "newSaveConfig": Object {},
  "saves": Object {},
  "syncError": undefined,
  "syncronizing": true,
  "uploadError": undefined,
  "uploading": false,
}
`;

exports[`handles syncronize actions reduces syncStart action 2`] = `
Object {
  "deleteError": undefined,
  "deleting": false,
  "newSaveConfig": Object {},
  "saves": Object {},
  "syncError": undefined,
  "syncronizing": true,
  "uploadError": undefined,
  "uploading": false,
}
`;

exports[`handles syncronize actions reduces syncSuccess action 1`] = `
Object {
  "deleteError": undefined,
  "deleting": false,
  "newSaveConfig": Object {},
  "saves": Object {
    "another_save": Object {
      "timestamp": 2234567,
    },
    "save 1": Object {
      "timestamp": 1234567,
    },
  },
  "syncError": undefined,
  "syncronizing": false,
  "uploadError": undefined,
  "uploading": false,
}
`;
